applications/wikipedia/main.py

```diff
@@ -100,17 +103,77 @@ def generate_batches(xs, batch_size=50):
     concurrency_limit=N_GPU,
     # Allow each container to process up to 10 batches at once.
     allow_concurrent_inputs=N_INPUTS,
+    secret=Secret.from_name("wandb"),
 )
 class TextEmbeddingsInference:
     def __enter__(self):
         # If the process is running for a long time, the client does not seem to close the connections, results in a pool timeout
+        import os
+        import uuid
+        from threading import Thread
+
         from httpx import AsyncClient
+        from pynvml import nvmlDeviceGetHandleByIndex, nvmlInit
+        from wandb import init
+        from wandb.sdk.wandb_run import Run
+
+        nvmlInit()
 
         self.process = spawn_server()
         self.client = AsyncClient(base_url="http://127.0.0.1:8000")
+        self.gpu = nvmlDeviceGetHandleByIndex(0)
+        self.track = True
+        # Check that user has configured
+        if CONFIGURE_LOGGING:
+            if not os.environ["WANDB_API_KEY"]:
+                print(
+                    "No Wandb API Key Configured -> No logging will be enabled for this run."
+                )
+                return
+            # We now generate a unique UUID for each run so that each run is not confused with one another
+            self.run: Run = init(project=WANDB_PROJECT_NAME, group=WANDB_GROUP_NAME)
```

Comment:
Hey Morgan, thanks for looking at the code. Mostly just style for me to be honest, I just prefer importing functions instead of calling it from the whole lib. Think I didn't get around to changing the second one down the line because I was trying to solve the async promise issue.

Does this change the behaviour of the library?